spec/datadog/core/environment/vm_cache_spec.rb

```diff
@@ -22,21 +22,67 @@
     describe '.global_constant_state' do
       subject(:global_constant_state) { described_class.global_constant_state }
 
-      it { is_expected.to be_a_kind_of(Integer) }
+      context 'on Ruby <= 3.1' do
+        before { skip('Test only runs on Ruby <= 3.1') if RUBY_VERSION >= '3.2.0' }
+
+        it { is_expected.to be_a_kind_of(Integer) }
+      end
+
+      context 'on Ruby >= 3.2' do
+        before { skip('Test only runs on Ruby >= 3.2') if RUBY_VERSION < '3.2.0' }
+
+        it { is_expected.to be nil }
+      end
     end
 
     describe '.global_method_state' do
       subject(:global_method_state) { described_class.global_method_state }
 
-      context 'with Ruby < 3', if: RUBY_VERSION < '3.0.0' do
+      context 'on Ruby 2.x' do
+        before { skip('Test only runs on Ruby 2.x') unless RUBY_VERSION.start_with?('2.') }
```

Comment:
Why replace `if: RUBY_VERSION < '3.0.0'` with the explicit skip?

I like the explicit skip when some test is not running, because it shows up in the rspec output that the testcase was skipped.

Otherwise, it's easy to do a change, run RSpec, it says all tests passed and you just broke some other Ruby version which is being skipped. Yes CI would catch it, but I still favor the more explicit style.

Reply:
I like this approach as well.